Services/LinkChecker/classes/class.ilLinkChecker.php

```diff
@@ -541,9 +541,10 @@ function __validateLinks($a_links)
 
 				$http_code = 0;
 				$c_error_no = 0;
+
+				$curl = new ilCurlConnection($link['complete']);
 				try
 				{
-					$curl = new ilCurlConnection($link['complete']);
```

Comment:
This might throw an ilCurlConnectionException. 
Therefore it should be catched, too.

Reply:
The check on line 510 should already avoid this exception:
```php
if(!ilCurlConnection::_isCurlExtensionLoaded())
{
	$this->__appendLogMessage('LinkChecker: Pear HTTP_Request is not installed. Aborting');
	ilLoggerFactory::getLogger('lchk')->error('LinkChecker: Curl extension is not loeaded. Aborting');
	return array();
}
```
However, this check might not be there forever.